src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -118,6 +130,15 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
         quiet.set(false)
         pull.set(false)
         cacheFrom.empty()
+        String safeTaskPath = path.replaceFirst("^:", "").replaceAll(":", "_")
+        imageIdFile.set(new File(project.buildDir, ".docker/${safeTaskPath}-imageId.txt"))
+
+        project.afterEvaluate {
```

Comment:
Heads up: We definitely don't want a `afterEvaluate`. It's also not required here as the Closure of `outputs.upToDateWhen` delays the evaluation.

Reply:
If the code runs outside the afterEvaluate, the class members such as url, certPath or apiVersion will not be set yet and the dockerClient might not represent the registry specified... So delaying the evaluation until the class is populated seems required to me.
Do you see some way around this?

Reply:
I already test cases for that so that should work as-is: https://github.com/bmuschko/gradle-docker-plugin/blob/master/src/functTest/groovy/com/bmuschko/gradle/docker/tasks/AbstractDockerRemoteApiTaskFunctionalTest.groovy#L87-L107.

Reply:
Ok removed the outputs.upToDateWhen and it seems the local variable have the time to get set. Did not know the upToDateWhen closure delayed evaluation.